.github/workflows/ci.yml

```diff
@@ -0,0 +1,161 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+env:
+  rust-version: 1.60.0
+
+jobs:
+  build:
+    name: cargo build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-build
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@v1
+        with:
+          toolchain: ${{ env.rust-version }}
+          target: wasm32-unknown-unknown
+      - name: Run builds
+        run: |
+          cargo build --workspace --exclude ic-cdk-optimizer --exclude ic-cdk-e2e-tests --target wasm32-unknown-unknown
+          cargo build --workspace --exclude ic-cdk-optimizer --exclude ic-cdk-e2e-tests --target wasm32-unknown-unknown --release
+
+  build-ic-cdk-optimizer:
+    name: Build ic-cdk-optimizer
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-optimizer
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@v1
+        with:
+          toolchain: ${{ env.rust-version }}
+      - name: Run builds
+        run: |
+          cargo build -p ic-cdk-optimizer
+
+  test:
+    name: cargo test
+    runs-on: ubuntu-latest
```

Comment:
Is there any reason to run this only on ubuntu-latest, rather than `runs-on: ${{ matrix.os }}` ?

Reply:
Crates in this repo (except for `ic-cdk-optimizer`) are for `--target wasm32-unknown-unknown`. So I thought run tests on multiple platforms doesn't provide that much benefit in test coverage. I just chose the fastest platform `ubuntu`.
A trade-off between CI time and test coverage.